ts/packages/ts/src/commands/commands.test.ts

```diff
@@ -221,5 +243,147 @@ describe(`commands`, () => {
+        expect.assertions(1);
+        await expect(() =>
+          new EvalCommand({
+            store: store1,
+            set: set1,
+            schema: new ConfigSchema('lazy', {
+              K1: {
+                type: 'String',
+                lazy: true,
+                required: true,
+              },
+              K2: {
+                type: 'String',
+              },
+            }),
+            configs: {
+              K2: '2',
+            },
+          }).run(),
+        ).rejects.toBeInstanceOf(ConfigError);
+      });
+      test('EvalCommand with override for lazy config when `cfgu.required = false`', async () => {
+        expect.assertions(1);
+        await expect(() =>
+          new EvalCommand({
+            store: store1,
+            set: set1,
+            schema: new ConfigSchema('lazy', {
+              K1: {
+                type: 'String',
+                lazy: true,
+                required: true,
+              },
+              K2: {
+                type: 'String',
+              },
+            }),
+            configs: {
+              K2: '2',
+            },
+          }).run(),
+        ).rejects.toBeInstanceOf(ConfigError);
+      });
+      test('run EvalCommand with override for lazy config when `cfgu.required = false`', async () => {
+        const result = await new EvalCommand({
+          store: store1,
+          set: set1,
+          schema: new ConfigSchema('lazy', {
+            K1: {
+              type: 'String',
+              lazy: true,
+              required: false,
+            },
+          }),
+          configs: {
+            K1: '1',
+          },
+        }).run();
+        expect(result).toMatchObject({ K1: { result: { value: '1' } } });
+      });
+      test('run EvalCommand with override for lazy config when `cfgu.required = true`', async () => {
+        const result = await new EvalCommand({
+          store: store1,
+          set: set1,
+          schema: new ConfigSchema('lazy', {
+            K1: {
+              type: 'String',
+              lazy: true,
+              required: true,
+            },
+          }),
+          configs: {
+            K1: '1',
+          },
+        }).run();
+        expect(result).toMatchObject({ K1: { result: { value: '1' } } });
+      });
+      test('run EvalCommand without override for lazy config when `cfgu.required = false`', async () => {
+        const result = await new EvalCommand({
+          store: store1,
+          set: set1,
+          schema: new ConfigSchema('lazy', {
+            K1: {
+              type: 'String',
+              lazy: true,
+            },
+          }),
+        }).run();
+        expect(result).toMatchObject({ K1: { result: { value: '' } } });
+      });
+
+      test('run EvalCommand with override for lazy config when there is a value in the store and get the override value in the result', async () => {
+        await store1.set([{ set: set1.path, key: 'K1', value: '2' }]);
```

Comment:
I think we need to avoid any direct use of the store methods from outside the commands themselves, we don't want people to get the idea that we want them to do these things, even for tests. Prefer using upsert.

Reply:
Strongly disagree.
This test has nothing to do we UpsertCommand. If it would be possible I would imitate a store with these in it.
Generally speaking, in uint-tests like in real life, one should eliminate all variables from the experiment and only change the tested one. proving two variables in one test is not valid for either. (unless they are entangled)

Reply:
Then at the very least remove the setup part from the test itself by using something like [`beforeAll`](https://jestjs.io/docs/setup-teardown#one-time-setup)

Reply:
done